packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1552,6 +1552,15 @@ abstract class InheritedWidget extends ProxyWidget {
   /// object.
   @protected
   bool updateShouldNotify(covariant InheritedWidget oldWidget);
+
+  /// The type of this inherited widget.
+  ///
+  /// Used by [BuildContext.inheritFromWidgetOfExactType] to find this widget.
+  ///
+  /// Defaults to [runtimeType], but subclasses can override to use another
+  /// type (e.g., the type of an object held by the subclass).
```

Comment:
should probably document that this absolutely under no circumstances may change during the lifetime of the object